presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -958,5 +959,11 @@ public PlanNode visitTableScan(TableScanNode node, RewriteContext<Expression> co
 
             return node;
         }
+
+        @Override
+        public PlanNode visitAssignUniqueId(AssignUniqueId node, RewriteContext<Expression> context)
+        {
+            return context.defaultRewrite(node, context.get());
```

Comment:
This doesn't look right. What if there's a predicate over the unique id column? It needs to be filtered out at this point.